docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+   - for **big** changes or backwards incompatible changes.
+
+### Devfile JSON Schema
+
+As mentioned above, the Devfile JSON schema is generated from the Go structs defined in the Devfile Kubernetes API. The latest JSON schema for a given API version is located under `schemas/<api-version>` in the [devfile/api repo](https://github.com/devfile/api/). 
+
+**Versioning Scheme**: Semver (major.minor.bugfix)
+
+**Update Process**: 
+  
+   1) Update the schema version string in the `// +devfile:jsonschema:version=<schema-version>` annotation in `pkg/apis/workspace/<k8s-api-version>/doc.go`
+   2) Re-generate the json schema
+   3) Publish new schema to devfile.io website
+
+**When to Update?** On each release of the schema, incremented based on the changes going in to the release (e.g. bugfix = comments / bugfixes, minor = larger and / or backwards compatible changes, major == breaking / backwards incompatible changes). K8S API version updates should also result in an appropriate increment of the schema version.
+
+
+### Relationship Between K8s API version and JSON Schema Version
+
+The Devfile JSON schema is generated from the Kubernetes API, and the version for the JSON schema is set in the doc.go file in the K8S API (`pkg/apis/workspace/<api-version>/doc.go).
+
+As we’re only updating the K8S API version when needed, but incrementing the schema version more frequently, this means that any given API version may point to multiple, backwards-compatible, schema versions over its lifespan. The schema version under `schemas/<api-version>` in [devfile/api repo](https://github.com/devfile/api/) points to the latest JSON schema generated from the K8S API.
+
+## Release Process
+The following steps outline the steps done to release a new version of the Devfile schema and publish its schemas to the devfile.io website
+
+   1) The release engineer tasked with creating the release clones the repository (and checks out the release branch if one already exists)
+
+   2) `make-release.sh <schema-version> <k8s-api-version>` is run:
+
+      i) A release branch (the name corresponding to the schema version) is created, if one does not already exist.
+
+      ii) The schema-version is updated in `pkg/apis/workspace/<api-version>/doc.go`.
+
+      iii) New JSON schemas are generated to include the new schema version
+      
+      iv) A new commit including the changes
+
+      v) Finally, a PR is opened to merge these changes into the release branch
+
+   3) Once the release PR is approved and merged, the release engineer creates a new release on GitHub based off the release branch that was just created and includes the generated `devfile.json` as a release artifact. 
+       - The tag `v{major}.{minor}.{bugfix}`, where the semver corresponds to the devfile schema version, is used to enable the new version of the API to be pulled in as a Go module.
```

Comment:
Change "semver" to "SemVer".

Reply:
Should we tag it with `v{major}.{minor}.{bugfix}` or just `{major}.{minor}.{bugfix}` (without the starting v)?

Reply:
`v{major}.{minor}.{bugfix}` is used here so that the tagged release can be pulled down as a Go module (which requires `v` at the beginning of a version)